src/cri_attributes.rs

```diff
@@ -99,12 +118,39 @@ pub(crate) mod parser {
             .map(|(i, extensions)| (i, ExtensionRequest { extensions }))
     }
 
-    fn parse_extension_request_ext(i: &[u8]) -> X509Result<ParsedCriAttribute> {
+    fn parse_extension_request_attr(i: &[u8]) -> X509Result<ParsedCriAttribute> {
         map(
             parse_extension_request,
             ParsedCriAttribute::ExtensionRequest,
         )(i)
     }
+
+    pub(super) fn parse_challenge_password(i: &[u8]) -> X509Result<ChallengePassword> {
+        // I'm sure, there is a more elegant way to try multiple parsers until the first succeeds,
+        // but I don't know nom well enough to implement it.
```

Comment:
The nom parser for this is [alt](https://docs.rs/nom/latest/nom/branch/fn.alt.html so I supposed the code below can be rewritten as something like `alt(parse_der_utf8string, parse_der_printablestring)`.
Example: https://github.com/rusticata/x509-parser/blob/f6d25f6bbcee178d143476752fb314d03662aa25/src/x509.rs#L153-L155